### PR TITLE
Modified the assemble and run script for better error handling

### PR DESCRIPTION
**Pull request by mik-dass**
Issue:https://github.com/redhat-developer/odo/issues/424
Signed-off-by: mdas <mrinald7@gmail.com>

### CODE REVIEW

File: scripts/assemble-and-restart.sh [hunk 1]
```diff
@@ -10,5 +13,9 @@ else
     /usr/libexec/s2i/assemble
 fi
 
+if [ $? != 0 ]; then
```
Comment (kadel) [inline scripts/assemble-and-restart.sh]: Is this condition still needed if we have `set -e`?

File: scripts/run.sh [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 # run s2i run script from source directory if it has been overridden  there
 if [ -f /opt/app-root/src/.s2i/bin/run ]; then
     exec /opt/app-root/src/.s2i/bin/run``
```
Comment (kadel) [inline scripts/run.sh]: Can you please remove `` from the end of this line?
It was added there before by mistake :-(

File: scripts/assemble-and-restart.sh [hunk 3]
```diff
@@ -10,5 +13,9 @@ else
...
+    exit 1
+fi
+
 # restart run process
-supervisorctl restart run
+supervisorctl restart run
```
Comment (kadel) [inline scripts/assemble-and-restart.sh]: please keep newline at the end of the file.

File: scripts/assemble-and-restart.sh (hunk 1 above)
Comment (mik-dass) [reply scripts/assemble-and-restart.sh]: OK removed

File: scripts/run.sh (hunk 2 above)
Comment (mik-dass) [reply scripts/run.sh]: Done

File: scripts/assemble-and-restart.sh (hunk 3 above)
Comment (mik-dass) [reply scripts/assemble-and-restart.sh]: Ok

**Comment by mik-dass** [comment]
@kadel Done

**Comment by kadel** [comment]
Thank you @mik-dass

**Pull request merged**